ros/src/common/cmake/autoware_build_flags/cmake/autoware_build_flags-extras.cmake

```diff
@@ -0,0 +1,5 @@
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
```

Comment:
These are only linker flags.  Are we not interested in setting CMAKE_CXX_FLAGS and CMAKE_C_FLAGS as well?

Reply:
I realise that adding something like CMAKE_CXX_FLAGS=Wall,Wextra,Werror may be too difficult to add into the code base now - there is probably quite a bit of work to do to get to the point where that is possible.

Reply:
Yeah, that'd be really useful. I mentioned that in https://github.com/CPFL/Autoware/pull/1385#issuecomment-406989958, but I think it'd be best to put those flags in a separate PR so that this one can be merged sooner. Also, I noticed that the flags are not consistent across all packages (e.g. `-O2` vs `-O3`, etc.), so for now I'd only add `-Wall` and the flags for enabling C++11.

Reply:
I've submitted #1395 which does that, though only for C++11